master/internal/db/postgres_trial_metrics.go

```diff
@@ -172,3 +316,35 @@ func GetMetrics(ctx context.Context, trialID, afterBatches, limit int,
 
 	return res, err
 }
+
+// shallowUnionMetrics unions non-overlapping AvgMetrics of two metrics bodies.
+func shallowUnionMetrics(oldBody, newBody *metricsBody) (*metricsBody, error) {
+	if oldBody == nil {
+		return newBody, nil
+	}
+
+	if newBody == nil {
+		return oldBody, nil
+	}
+
+	// FIXME: We don't need to convert this
+	oldAvgMetricsMap := oldBody.AvgMetrics.AsMap()
+	newAvgMetricsMap := newBody.AvgMetrics.AsMap()
+	for key, newValue := range newAvgMetricsMap {
+		// we cannot calculate min/max efficiently for replaced metric values
+		// so we disallow it.
+		if _, ok := oldAvgMetricsMap[key]; ok {
+			fmt.Println("old, new", oldBody.AvgMetrics, newBody.AvgMetrics)
+			return nil, errors.Errorf("replacing existing metric values is not implemented %s", key)
```

Comment:
```suggestion
			return nil, errors.Errorf("overwriting existing metrics is not supported, conflicting key: %s", key)
```

Reply:
also we've been trying to start using `fmt.Errorf` instead `errors.Errorf` or `errors.Wrap`... but I guess that effort is in about the same state as replacing queryprotof with bun.

Reply:
I updated both. IIRC we had a linter on what error package we should be using. is that still in somewhere?

Reply:
We have a linter that disallows ```errors``` right now to prefer ```"github.com/pkg/errors"``` (which seems somewhat backwards since ```"github.com/pkg/errors"``` doesn't have maintainers / is in read only mode iirc)

I got half way with a linter banning ```errors.Wrap``` but a lot of the code base does patterns like 

```
return obj, errors.Wrapf(queryThisThing, "this returns nil when the error is nil"
```

which made me question the time trade off vs value of everything using ```fmt.Errorf("%w")```. I tried to automate it and I think it was like a 2k+ line PR and it would have been really hard to get someone to review that.

I think you have mentioned it before but would be cool to have a linter that could disallow new code to allow us to automate coding standards that are right now sometimes done in code review by only a subset of people.

Reply:
> I think you have mentioned it before but would be cool to have a linter that could disallow new code to allow us to automate coding standards that are right now sometimes done in code review by only a subset of people.

as in a way to introduce a linter with it being able to selectively ignore old/existing complaints? Yeah that'd be great.
I did a dumb version of this here https://github.com/determined-ai/determined/pull/6414/commits/17ae66aef7ff2ff7226a2a7cb8f3088d45b8e1f8 which is based on file boundaries. I've seen Codeclimate do something similar to what we want, they have might have tooling around it. running the linter twice on two revisions and diffing the complaints might be one way too 🤔